great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -1733,10 +1733,31 @@ def _pandas_map_condition_index(
 
     df = df[boolean_mapped_unexpected_values]
 
-    if result_format["result_format"] == "COMPLETE":
-        return list(df.index)
+    if "unexpected_index_column_names" in result_format:
+
+        unexpected_index_list: Optional[List[Dict[str, Any]]] = []
+        unexpected_index_column_names: List[str] = result_format[
+            "unexpected_index_column_names"
+        ]
+        unexpected_indices: List[int] = list(df.index)
+        for index in unexpected_indices:
+            primary_key_dict: Dict[str, Any] = {}
+            for column_name in unexpected_index_column_names:
+                if column_name not in metrics["table.columns"]:
+                    raise ge_exceptions.InvalidMetricAccessorDomainKwargsKeyError(
+                        message=f'Error: The unexpected_index_column: "{column_name}" in does not exist in Dataframe. '
+                        f"Please check your configuration and try again."
+                    )
+                primary_key_dict[column_name] = df.at[index, column_name]
+            unexpected_index_list.append(primary_key_dict)
+        if result_format["result_format"] == "COMPLETE":
+            return unexpected_index_list
+        return unexpected_index_list[: result_format["partial_unexpected_count"]]
```

Comment:
Added the filtering for `partial_unexpected_count` for both `unexpected_index_column` and just `unexpected_index` (existing behavior)